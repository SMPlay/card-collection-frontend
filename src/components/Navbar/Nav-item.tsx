import React from "react";
import { createUseStyles, useTheme } from "react-jss";

import arrowImg from "../../img/arrow.png";

const useStyles = createUseStyles({
  navItem: {
    position: "relative",
    fontSize: 22,
    fontWeight: 600,
    width: 150,
    height: 27,
    padding: "0 10px",
    color: "rgb(255, 255, 255)",
    background: ({theme}) => theme.colorPrimary,
    transform: "skew(-20deg)",
    cursor: "pointer",
    border: "1px solid rgb(18, 18, 18)",
    "&::after": {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      content: "''",
      width: "100%",
      height: "100%",
      opacity: 0,
      background: "linear-gradient(to top, rgba(255, 255, 255, .5), transparent)",
      transition: "opacity .2s ease",
    },
    "&:hover": {
      "&::after": { opacity: .6 }
    }
  },
  navItemChild: {
    position: "relative",
    zIndex: 100,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    transform: "skew(20deg)",
  },
  arrow: {
    position: "absolute",
    top: 5,
    right: -7,
    transform: "rotate(-90deg)",
    transition: "transform .2s ease",
    "&.open": {
      transform: "rotate(0)"
    }
  },
});

interface NavItemProps {
  isManyThings?: boolean;
  isOpen?: boolean;
}

export const NavItem: React.FC<NavItemProps> = ({ children, isManyThings, isOpen }) => {
  const theme = useTheme();
  const styles = useStyles({theme});

  return (
    <div className={styles.navItem}>
      <span className={styles.navItemChild}>
        {children}
        {isManyThings && 
        <img 
          src={arrowImg} 
          alt="right arrow" 
          className={`${styles.arrow} ${isOpen ? "open" : ""}`}/>}
      </span>
    </div>
  );
}
