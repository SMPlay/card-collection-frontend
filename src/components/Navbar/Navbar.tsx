import React from "react";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import { Link } from "react-router-dom";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

import { PageType } from "../../types/PageType";

const useStyles = makeStyles((theme: Theme) => 
  createStyles({
    navlink: { color: "rgb(79, 79, 79)" }
  })
);

interface NavbarProps {
  pages: PageType[];
}

export const Navbar: React.FC<NavbarProps> = ({ pages }) => {
  const classes = useStyles();

  return (
    <List>
      {pages.map((page) => (
        <Link
          to={page.url}
          key={page.pageName}>
          <ListItem button>
            <ListItemText classes={{primary: classes.navlink}} primary={page.pageName} />
          </ListItem>
        </Link>
      ))}
    </List>
  );
};
