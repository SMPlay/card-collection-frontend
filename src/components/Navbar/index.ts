export { Navbar } from "./Navbar";
export { NavItem } from "./Nav-item";
