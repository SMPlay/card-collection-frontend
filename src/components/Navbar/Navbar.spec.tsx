import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import { Navbar } from "./Navbar";

describe("Header component", () => {
  it("should render", () => {
    const tree = renderer.create(
      <Router>
        <Navbar/>
      </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});