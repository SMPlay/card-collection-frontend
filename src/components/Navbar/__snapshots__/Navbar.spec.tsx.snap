// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-1 MuiTypography-body1 MuiTypography-displayBlock"
        >
          Главная
        </span>
      </div>
    </div>
  </a>
  <a
    href="/collections"
    onClick={[Function]}
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-1 MuiTypography-body1 MuiTypography-displayBlock"
        >
          Коллекции
        </span>
      </div>
    </div>
  </a>
  <a
    href="/gallery"
    onClick={[Function]}
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-1 MuiTypography-body1 MuiTypography-displayBlock"
        >
          Галерея
        </span>
      </div>
    </div>
  </a>
  <a
    href="/auction"
    onClick={[Function]}
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-1 MuiTypography-body1 MuiTypography-displayBlock"
        >
          Аукцион
        </span>
      </div>
    </div>
  </a>
</ul>
`;
